### ISSUE TITLE
Could not parse repo URL

### ISSUE THREAD

**Issue by ragnarpa** [argoproj/argo-cd#9440]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

SSH URL parsing of GitHub repos is failing.

<br class="Apple-interchange-newline">

**To Reproduce**

1. Install Argo CD 2.4 RC
2. Add a Git repo with SSH URL
3. Deploy applications that reference the added repo
4. Monitor Argo CD repo server logs
5. You should see warnings like following

```
time="2022-05-18T08:06:28Z" level=warning msg="Could not parse repo URL 'git@github.com:my-account/my-repo.git': parse \"git@github.com:my-account/my-repo.git\": first path segment in URL cannot contain colon"
```

**Expected behavior**

No warnings in the logs. Current warnings are confusing and leave a user with an impression as if something is broken.

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

v2.4.0+68a0a83

**Logs**

```
Paste any relevant application logs here.
```

```
time="2022-05-18T08:06:28Z" level=warning msg="Could not parse repo URL 'git@github.com:my-account/my-repo.git': parse \"git@github.com:my-account/my-repo.git\": first path segment in URL cannot contain colon"
```

**Comment by booleanbetrayal** [comment]
Just a confirmation that we are seeing this as well in our logs.

**Comment by qixiaobo** [comment]
The same problem

**Comment by hasland** [comment]
Our ArgoCD server is showing this logs too but it doesn't look just a warning for us because our repos connection fail to connect from time to time. We can see it in Argo config > repositories - from time to time some repos got Failed status in Connection Status, then after sometime it connects back and change the status to Successfull

**Comment by mars64** [comment]
tldr; I hit this issue, and it was related to node egress security groups. 

I ran into this issue using versions of argocd that previously worked in other clusters (`2.4.5`-`2.4.8`), and was confused why suddenly I started seeing this issue on `2.4.8`. As it happened, a colleague had constructed an incomplete EKS configuration using the `terraform-aws-eks` provider -- by default, the provider only adds security groups for control plane communication -- no node-to-node or egress. By [implementing the basic security groups as per the complete example](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/complete/main.tf#L67), I was able to resolve this behavior.

Confirmed on at least ArgoCD `v2.4.8+844f79e`

**Comment by slamer59** [comment]
Lose 4h... to see that it works even with the error message ...

**Comment by qixiaobo** [comment]
> Lose 4h... to see that it works even with the error message ...

Time killer 😢

**Comment by kenchan0130** [comment]
I have already made tentative corrections [here](https://github.com/argoproj/argo-cd/pull/9836).
I am waiting for your vote as it seems to be a low priority for review.

**Comment by crenshaw-dev** [comment]
Thanks for y'alls patience! I just merged and cherry-picked the fix. I'll see if folks are okay with a release of 2.4.10 today.
